wcc-contentful/README.md

```diff
@@ -0,0 +1,229 @@
+```
+
+See the {WCC::Contentful::Model} documentation for more details.
+
+### Store API
+
+The Store layer is used by the Model API to access Contentful data in a raw form.
+The Store layer returns entries as hashes parsed from JSON, conforming to the
+object structure returned from the Contentful CDN.
+
+The following examples show how to use the Store API to retrieve raw data from
+the store:
+
+```ruby
+store = WCC::Contentful::Services.instance.store
+# => #<WCC::Contentful::Store::CDNAdapter:0x00007fb92a221498
+
+store.find('5FsqsbMECsM62e04U8sY4Y')
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+
+store.find_by(content_type: 'page', filter: { slug: '/some-slug' })
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+
+query = store.find_all(content_type: 'page').eq('group', 'some-group')
+# => #<WCC::Contentful::Store::CDNAdapter::Query:0x00007fa3d40b84f0
+query.first
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+query.result
+# => #<Enumerator::Lazy: ...>
+query.result.force
+# => [{"sys"=> ...}, {"sys"=> ...}, ...]
+```
+
+See the {WCC::Contentful::Store} documentation for more details.
+
+### Direct CDN API (SimpleClient)
+
+The SimpleClient is the bottom layer, and is used to get raw data directly from
+the Contentful CDN.  It handles response parsing and paging, but does not resolve
+links or transform the result into a Model class.
+
+The following examples show how to use the SimpleClient to retrieve data directly
+from the Contentful CDN:
+
+```ruby
+client = WCC::Contentful::Services.instance.client
+# => #<WCC::Contentful::SimpleClient::Cdn:0x00007fa3cde89310
+
+response = client.entry('5FsqsbMECsM62e04U8sY4Y')
+# => #<WCC::Contentful::SimpleClient::Response:0x00007fa3d103a4e0
+response.body
+# => "{\n  \"sys\": {\n ...
+response.raw
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+
+client.asset('5FsqsbMECsM62e04U8sY4Y').raw
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+
+response = client.entries('fields.group' => 'some-group', 'limit' => 5)
+# => #<WCC::Contentful::SimpleClient::Response:0x00007fa3d103a4e0
+response.count
+# => 99
+response.first
+# => {"sys"=>
+#  ...
+# "fields"=>
+# ...}
+response.items
+=> #<Enumerator::Lazy: ...>
+response.items.count  # Careful! This evaluates the lazy iterator and gets all pages
+# => 99
+
+response.includes
+# => {"4xNnFJ77egkSMEogE2yISa"=>
+#   {"sys"=> ...}
+#  "6Fwukxxkxa6qQCC04WCaqg"=>
+#   {"sys"=> ...}
+#   ...}
+```
+
+The client handles Paging automatically within the lazy iterator returned by #items.
+This lazy iterator does not respect the `limit` param - that param is only passed
+through to the API to set the page size.
+
+Entries included via the `include` parameter are made available on the #includes
+field.  This is a hash of `<entry ID> => <raw entry>` and makes it easy to grab
+links.  This hash is added to lazily as you enumerate the pages.
+
+See the {WCC::Contentful::SimpleClient} documentation for more details.
+
+### Accessing the APIs within application code
```

Comment:
should this move to the `app` gem?

Reply:
The Model API is still inside the `wcc-contentful` base gem, so this documentation is still valid here.  What got moved into `-app` is the validations, and our own pre-defined overrides of `WCC::Contentful::Model::Page`